org.bndtools.launch/src/bndtools/launch/AbstractOSGiLaunchDelegate.java

```diff
@@ -171,70 +182,153 @@ public boolean buildForLaunch(ILaunchConfiguration configuration, String mode, I
 	@Override
 	public boolean finalLaunchCheck(ILaunchConfiguration configuration, String mode, IProgressMonitor monitor)
 		throws CoreException {
-
 		try {
 			run = LaunchUtils.createRun(configuration, getRunMode());
+			return Central.getWorkspace()
+				.readLocked(() -> {
+
+					initialiseBndLauncher(configuration, run);
+
+					if (!checkProjectErrors(configuration, mode, monitor))
+						return false;
+
+					if (prefs.getWarnExistingLaunches() && !checkExistingLaunches(configuration))
```

Comment:
There has been a subtle change here: The call here to `prefs.getWarnExistingLaunches()` is referencing the field called `prefs`. Before your changes, `finalLaunchCheck()` would instantiate a new local `BndPreferences` called `prefs` and use that. If I read the code correctly, this should not change the behaviour as `BndPreferences` doesn't cache the results, but reads them on-demand from the preference store. So this is probably an improvement.